File: 2022/1/solve.js
const {readFileSync, promises: fsPromises} = require('fs');
const contents = readFileSync("input.txt", "utf-8");
const arr = contents.split(/\r?\n/);


let m = 0
let sum_curr = 0

for (let i=0;i<arr.length;i++) {
  curr = arr[i]
  // If new line
  if (curr != "") {
    sum_curr += parseInt(curr)
  // If Int
  } else {
    if (sum_curr > m) {
      m = sum_curr
    }
    sum_curr = 0
  }
}


console.log(m)

File: 2022/1/solve.rs
use std::fs;

fn main() {
  let content = fs::read_to_string("input.txt")
    .expect("IDK");
  let arr: Vec<&str> = content.split("\n").collect();

  let mut score = 0;
  let mut max_score = 0;

  for i in arr {
    if i.eq("") {
      if max_score < score {max_score = score}
      score = 0
    } else {
      score += i.parse::<i32>().unwrap();
    } 
  }
  println!("{max_score}");
}

